### PR TITLE
Add implementation plan

### PR DESCRIPTION
**Pull request by dseefeld**

### CODE REVIEW

File: Documentation/planning/arcade-powered-source-build/implementation-plan.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Source-build 5.0 Implementation Plan
+
+To get each repo building with the new source-build 5.0 plan, [Arcade-Powered Source-Build](https://github.com/dotnet/source-build/tree/release/3.1/Documentation/planning/arcade-powered-source-build), the following stages will be accomplished.  Below, the repos are placed in a table representing the earliest dependencies are available and work can begin.  Stages can be completed in parallel, but stage 4 may depend on output from stage 6 from previous repos.  The following graph and table will be used to track progress for each repo.
+
+> ![](img/implementation-plan-graph.png)
+> [source (img/implementation-plan-graph.dgml)](img/implementation-plan-graph.dgml)
+
+| Tier | Repo | (Stage 1) Build from source - 5.0 | (Stage 2) Patch Removal | (Stage 3) Move build params to eng dir | (Stage 4) Input Intermediate Packages Available | (Stage 5) Official build | (Stage 6) Source-build PR validation | (Stage 7) Prebuilt check / enforcement |
```
Comment (dagood) [inline Documentation/planning/arcade-powered-source-build/implementation-plan.md:8]: Simplify stage names for less distracting wrapping with GitHub's width limit?

```suggestion
| Tier | Repo | (1) Build from source - 5.0 | (2) Patch Removal | (3) Move build params to eng dir | (4) Input Intermediate Packages Available | (5) Official build | (6) Source-build PR validation | (7) Prebuilt check / enforcement |
```

File: Documentation/planning/arcade-powered-source-build/implementation-plan.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+| 4 | vstest |  | 6 | | | | | |
+| 4 | fsharp |  | 2 | | | | | |
+| 5 | installer |  | 5 | | | | | |
+
+## Stage descriptions:
+  - **(1) Build from Source 5.0** – Get repo building from source with 5.0 source
+  - **(2) Patch Removal** – Incorporate patches required to build from source into the repo code base.
+  - **(3) Move Build Params to Eng Dir** – All source-build specific build parameter from repos/<reponame>.proj moved from source-build repo to repo /eng directory.
+  - **(4) Input Intermediate Packages Available** – Indicates upstream intermediate packages are available, and work on this repo can begin.
+  - **(5) Official build** – Repo builds in official build, producing source-built intermediate packages.
```
Comment (dagood) [inline Documentation/planning/arcade-powered-source-build/implementation-plan.md:40]: Action:

```suggestion
  - **(5) Official build** – Build source-built intermediate packages in official build.
```

File: Documentation/planning/arcade-powered-source-build/implementation-plan.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  - **(6) Source-build PR validation** – Azdo build jobs are added to ensure build-from-source continues running
```
Comment (dagood) [inline Documentation/planning/arcade-powered-source-build/implementation-plan.md:41]: Action:

```suggestion
  - **(6) Source-build PR validation** – Add AzDO build jobs to PR validation to ensure build-from-source continues running without avoidable regressions.
```

File: Documentation/planning/arcade-powered-source-build/implementation-plan.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+| 1 | newtonsoft-json | X | - | | | | | |
+| 1 | netcorecli-fsc | X | - | | | | | |
+| 1 | newtonsoft-json901 | X | - | | | | | |
+| 1 | xliff-tasks | X | 1 | | | | | |
+| 1 | clicommandlineparser | X | 1 | | | | | |
+| 1 | roslyn | X | 4 | | | | | |
+| 2 | linker | X | 3 | | | | | |
+| 2 | runtime | X | 10 | | | | | |
+| 2 | msbuild | X | 4 | | | | | |
+| 2 | nuget-client | X | 8 | | | | | |
```
Comment (dagood) [inline Documentation/planning/arcade-powered-source-build/implementation-plan.md:24]: Repo name seems better for this doc vs. sb proj file name:

```suggestion
| 2 | NuGet.Client | X | 8 | | | | | |
```

File: Documentation/planning/arcade-powered-source-build/implementation-plan.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+| 5 | installer |  | 5 | | | | | |
+
+## Stage descriptions:
+  - **(1) Build from Source 5.0** – Get repo building from source with 5.0 source
+  - **(2) Patch Removal** – Incorporate patches required to build from source into the repo code base.
+  - **(3) Move Build Params to Eng Dir** – All source-build specific build parameter from repos/<reponame>.proj moved from source-build repo to repo /eng directory.
+  - **(4) Input Intermediate Packages Available** – Indicates upstream intermediate packages are available, and work on this repo can begin.
+  - **(5) Official build** – Repo builds in official build, producing source-built intermediate packages.
+  - **(6) Source-build PR validation** – Azdo build jobs are added to ensure build-from-source continues running
+  - **(7) Prebuilt check / enforcement** – Adds a check to PR validation to ensure that no prebuilts are included when building repo
```
Comment (dagood) [inline Documentation/planning/arcade-powered-source-build/implementation-plan.md:42]:
```suggestion
  - **(7) Prebuilt check / enforcement** – Add a check to PR validation to ensure that no prebuilts are included when building repo.
```

File: Documentation/planning/arcade-powered-source-build/implementation-plan.md (hunk 1 above)
Comment (dseefeld) [reply Documentation/planning/arcade-powered-source-build/implementation-plan.md:8]: I just changed them to include "stage" to better match the legend on the graph.  I'm going to try multi-line headers to see if this helps.

**Pull request merged**